components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -3436,52 +3427,97 @@ public void deleteAPI(API api) throws APIManagementException {
                                 + " successfully removed from the database.";
                 log.debug(logMessage);
             }
+        } catch (APIManagementException e) {
+            log.error("Error while executing API delete operations on DB for API " + api.getUuid(), e);
+            isError = true;
+        }
 
-            JSONObject apiLogObject = new JSONObject();
-            apiLogObject.put(APIConstants.AuditLogConstants.NAME, api.getId().getApiName());
-            apiLogObject.put(APIConstants.AuditLogConstants.VERSION, api.getId().getVersion());
-            apiLogObject.put(APIConstants.AuditLogConstants.PROVIDER, api.getId().getProviderName());
-
-            APIUtil.logAuditMessage(APIConstants.AuditLogConstants.API, apiLogObject.toString(),
-                    APIConstants.AuditLogConstants.DELETED, this.username);
+        JSONObject apiLogObject = new JSONObject();
+        apiLogObject.put(APIConstants.AuditLogConstants.NAME, api.getId().getApiName());
+        apiLogObject.put(APIConstants.AuditLogConstants.VERSION, api.getId().getVersion());
+        apiLogObject.put(APIConstants.AuditLogConstants.PROVIDER, api.getId().getProviderName());
 
-            cleanUpPendingAPIStateChangeTask(apiId);
+        APIUtil.logAuditMessage(APIConstants.AuditLogConstants.API, apiLogObject.toString(),
+                APIConstants.AuditLogConstants.DELETED, this.username);
```

Comment:
shall we move this to just after priority deletion tasks are completed?
we can do it just before
```
        if (apiId != -1) {
            try {
                cleanUpPendingAPIStateChangeTask(apiId);
```